### PR TITLE
feat(design): explode complex types

### PR DESCRIPTION
**Pull request by kpanot**
## Proposed change

Explode complex types to generate "sub-Token" for each field of the complex type

## Related issues

*- No issue associated -*

### CODE REVIEW

File: packages/@o3r/design/src/core/design-token/parsers/design-token.parser.ts [hunk 1]
```diff
@@ -161,17 +161,49 @@ const getCssRawValue = (variableSet: DesignTokenVariableSet, token: DesignTokenV
   }
 };
 
+const convertOjectToToken = (node: any): DesignTokenGroup | undefined => {
```
Comment (cpaulve-1A) [inline packages/@o3r/design/src/core/design-token/parsers/design-token.parser.ts:164]:
```suggestion
const convertObjectToToken = (node: any): DesignTokenGroup | undefined => {
```

File: packages/@o3r/design/src/core/design-token/parsers/design-token.parser.ts [hunk 2]
```diff
@@ -161,17 +161,49 @@ const getCssRawValue = (variableSet: DesignTokenVariableSet, token: DesignTokenV
...
+  if (typeof node !== 'object') {
+    return;
+  }
+
+  if (Array.isArray(node)) {
+    return node
+      .map((item, index): DesignTokenGroup => ({ [index]: convertOjectToToken(item) }))
```
Comment (cpaulve-1A) [inline packages/@o3r/design/src/core/design-token/parsers/design-token.parser.ts:171]:
```suggestion
      .map((item, index): DesignTokenGroup => ({ [index]: convertObjectToToken(item) }))
```

File: packages/@o3r/design/src/core/design-token/parsers/design-token.parser.ts [hunk 3]
```diff
@@ -161,17 +161,49 @@ const getCssRawValue = (variableSet: DesignTokenVariableSet, token: DesignTokenV
...
+
+/**
+ * Generate a new Token for each field of a complex type
+ * @param node Node of the Token in the parsed token files
+ */
+const explodeComplexType = (node: DesignTokenNode): DesignTokenGroup | undefined => {
+  if (typeof node.$value !== 'object') {
+    return;
+  }
+  return convertOjectToToken(node.$value);
```
Comment (cpaulve-1A) [inline packages/@o3r/design/src/core/design-token/parsers/design-token.parser.ts:192]:
```suggestion
  return convertObjectToToken(node.$value);
```

File: packages/@o3r/design/src/core/design-token/design-token-specification.interface.ts [hunk 4]
```diff
@@ -223,26 +227,26 @@ export interface DesignTokenGroupCommonFields<G extends DesignTokenExtensions> {
...
+  | DesignTokenTypeNumber
+  | DesignTokenTypeStrokeStyle
+  | DesignTokenTypeBorder
+  | DesignTokenTypeTransition
+  | DesignTokenTypeShadow
+  | DesignTokenTypeGradient
+  | DesignTokenTypeTypography
+  | DesignTokenTypeImplicit;
+
 /** Available Design Token types */
```
[the remark this answers is not in the document]
Comment (kpanot) [reply packages/@o3r/design/src/core/design-token/design-token-specification.interface.ts:248]: Indeed...
I added explanations and note (and remove the export of this technical interface)

**Review by cpaulve-1A — APPROVED** [review]

**Review by fpaul-1A — APPROVED** [review]

**Pull request merged**